### PR TITLE
Feat(eligibility/form): custom validation message for index and confirm fields

### LINKED ISSUE #2012: Form field validation: update copy to be more specific on the problem
image

image

## Acceptance Criteria

- [x] Configure the browser validation message - https://developer.mozilla.org/en-US/docs/Web/HTML/Constraint_validation#controlling_the_text_of_constraint_violation
- [x] Test with different browser locales
- [x] Add custom copy for both Eligibility Index (radio buttons) and Eligibility Confirm (text fields)

### PR DESCRIPTION
**Pull request by angela-tran**
Closes #2012

### Eligibility Index form
1. Don't select any transit benefit
2. Click "Choose this benefit"
3. You should get a browser validation message that says "Please choose a transit benefit."

### Eligibility Confirm form
**Cases to try with `sub` field (card number field):**
 - Enter not enough digits
 - Enter non-numeric characters
 - Leave the field blank

In all those cases, you should get a browser validation message that says "Please enter a X-digit number." where X is the number specific for the transit benefit (5 for MST Courtesy Card, 4 for SBMTD Reduced Fare Mobility ID). See https://github.com/cal-itp/benefits/issues/2012#issuecomment-2071090041 for more context.

**Cases to try with `name` field (last name field):**
 - Enter more than 255 characters
 - Leave the field blank

In all those cases, you should get a browser validation message that says "Please enter your last name."

## Screenshots
<details><summary>Expand</summary>

image

image
</details>

### CODE REVIEW

**Comment by angela-tran** [comment]
Side note: our Cypress tests were actually really helpful for this PR. They fail tests that encounter syntax errors, which caught regressions on pages that contain forms but don't need custom validation messages. 🎉 

See GitHub Action runs:
https://github.com/cal-itp/benefits/actions/runs/8760017173/job/24044225079
https://github.com/cal-itp/benefits/actions/runs/8790692524/job/24123260719
https://github.com/cal-itp/benefits/actions/runs/8791452896/job/24125714216

File: benefits/core/templates/core/includes/form.html [hunk 1]
```diff
@@ -59,7 +59,29 @@
         $("#{{ form.id }}").on("submit", function(e) {
           $(this).trigger("submitting");
         });
-      });
+
+        {% if use_custom_validity %}
```
Comment (thekaveman) [inline benefits/core/templates/core/includes/form.html:63]: I'm curious if there is another way to derive whether or not this Javascript should be generated here, besides a new context variable? I'm thinking it's one more thing to remember to add when we want this behavior on forms. Is it possible to use the presence of the custom messages or something else?

Comment (thekaveman) [reply benefits/core/templates/core/includes/form.html:63]: Like maybe a property on the form itself?

Comment (angela-tran) [reply benefits/core/templates/core/includes/form.html]: Does https://github.com/cal-itp/benefits/pull/2045/commits/59c4808ff7fa7dd61909b802eb1d019b6eff1d4f look ok?

**Review by thekaveman — CHANGES_REQUESTED** [review]
Yes that looks good, just one more small change.

File: benefits/eligibility/forms.py [hunk 2]
```diff
@@ -127,6 +130,8 @@ def __init__(
         name_widget = widgets.FormControlTextInput(placeholder=name_placeholder)
         if name_max_length:
             name_widget.attrs.update({"maxlength": name_max_length})
+        if name_custom_validity:
+            name_widget.attrs.update({"data-custom-validity": name_custom_validity})
```
[the remark this answers is not in the document]
Comment (angela-tran) [reply benefits/eligibility/forms.py:134]: Yes, thanks for catching that. Fixed in f6ebcd13

**Comment by angela-tran** [comment]
Need to resolve conflicts

File: benefits/eligibility/forms.py [hunk 3]
```diff
@@ -30,8 +30,11 @@ def __init__(self, agency: models.TransitAgency, *args, **kwargs):
         self.classes = "col-lg-8"
         # second element is not used since we render the whole label using selection_label_template,
         # therefore set to None
-        self.fields["verifier"].choices = [(v.id, None) for v in verifiers]
-        self.fields["verifier"].widget.selection_label_templates = {v.id: v.selection_label_template for v in verifiers}
+        verifier_field = self.fields["verifier"]
+        verifier_field.choices = [(v.id, None) for v in verifiers]
+        verifier_field.widget.selection_label_templates = {v.id: v.selection_label_template for v in verifiers}
+        verifier_field.widget.attrs.update({"data-custom-validity": _("Please choose a transit benefit.")})
```
Comment (thekaveman) [inline benefits/eligibility/forms.py:36]: Oops, I don't think this string `"Please choose a transit benefit."` made its way into the `.po` files?

**Review by thekaveman — CHANGES_REQUESTED** [review]

File: benefits/eligibility/forms.py (hunk 3 above)
Comment (angela-tran) [reply benefits/eligibility/forms.py:36]: Good catch, thanks -- I added it in new commit https://github.com/cal-itp/benefits/pull/2045/commits/ead179bb898a118e6cca7f03adad742bd65499c0

**Comment by angela-tran** [comment]
Rebased on top of #2042 and waiting for it to be merged to minimize merge conflicts

**Review by thekaveman — APPROVED** [review]
## Eligibility index

Confirming custom error messages:

English (mobile) | Spanish (desktop - w/local edits to `.po`)
-----|------
image | image

## Eligibility confirm

Confirming custom error messages:

Format | English (mobile) | Spanish (desktop w/local edits to `.po`)
-------|------------------|-----------------------------------------
`sub` missing MST | image | image
`sub` missing SBMTD | image | image
`sub` length MST | image | image
`sub` length SBMTD | image | image
`sub` mismatch MST | image | image
`sub` mismatch SBMTD | image | image
`name` missing MST | image | image
`name` missing SBMTD | image | image

**Pull request merged**